server/src/utils/database.ts

```diff
@@ -58,12 +58,18 @@ export function searchAssetBuilder(
     builder.andWhere(`${builder.alias}.ownerId IN (:...userIds)`, { userIds: options.userIds });
   }
 
-  const path = _.pick(options, ['encodedVideoPath', 'originalPath', 'previewPath', 'thumbnailPath']);
+  const path = _.pick(options, ['encodedVideoPath', 'previewPath', 'thumbnailPath']);
   builder.andWhere(_.omitBy(path, _.isUndefined));
 
   if (options.originalFileName) {
     builder.andWhere(`f_unaccent(${builder.alias}.originalFileName) ILIKE f_unaccent(:originalFileName)`, {
-      originalFileName: `%${options.originalFileName}%`,
+      originalFileName: options.originalFileName.replaceAll('*', '%'),
+    });
+  }
+
+  if (options.originalPath) {
```

Comment:
I don't think we have this in the UI

Reply:
Not yet, should I add it? It's relevant for existing folder structures when using external libraries.

Reply:
I don't think this has an index so it probably won't work very well as is.

Reply:
Is there a need to handle path and filename separately? It seems to me that using `originalPath` would work for both.

Reply:
For non-external assets they are different. Or, depending on you storage template they can be completely independent.

Reply:
> No, we plan to offer these as advanced free text searches in which we include some type of DSL in the input text. So probably it makes sense to have us implement this in the future altogether

A single text input where I can enter something like `path:"*/Paris 2022/*"`? That sounds great, I guess I can close this ticket then.